docs/dev/policy_documents/default-build-flags.md

```diff
@@ -0,0 +1,194 @@
+
+```
+> time /hab/pkgs/ssd/bc/1.07.1/20200427105511/bin/bc -l <<< "scale=4096;4*a(1)"
+...SNIP...
+real	0m10.950s
+user	0m10.960s
+sys	0m0.004s
+
+```
+
+This example was purposefully constructed to show a large speedup from
+optimization. However, I expect that the majority of software written
+in C in core-plans would benefit from ensuring we create optimized
+builds.
+
+Some software in core-plans already builds with optimizations, either
+because their build scripts add the required flags or because the plan
+we created does.
+
+However, it is common practice for distributions to set a default
+level of optimization for all builds. This is in part because some
+security features are also tied to having optimization enabled:
+
+For example,
+
+- [Omnibus: -02](https://github.com/chef/omnibus/blob/0baf021d62d9e61687e2d7efbac865a9640027c2/lib/omnibus/software.rb#L753)
+- [NixPkgs: -02 enabled when hardening is enabled (which is the default)](https://github.com/NixOS/nixpkgs/blob/4707dc6454904bee0bacd3a53829bcc76b4f0667/pkgs/build-support/cc-wrapper/add-hardening.sh#L41)
+- [Debian: -02](https://manpages.debian.org/testing/dpkg-dev/dpkg-buildflags.1.en.html)
+
+## Security
+
+Compilers have implemented a broad range of security
+mitigations. These include things like stack protectors that help
+prevent security bugs because of buffer overflows, automatic rewriting
+of unsafe libc functions into safer variants, and position independent
+executables to support address space layout randomization which makes
+various exploits harder to write.
+
+These mitigations are enabled via compiler and linker flags.  Most
+distributions have made such flags their default:
+
+https://fedoraproject.org/wiki/Changes/Harden_All_Packages
+https://wiki.debian.org/HardeningWalkthrough
+https://nixos.org/nixpkgs/manual/#sec-hardening-in-nixpkgs
+https://wiki.gentoo.org/wiki/Project:Hardened
+
+## Why a Policy?
+
+Multiple users have attempted to make progress on this issue in the
+past. For example,
+
+- https://github.com/habitat-sh/core-plans/issues/587
+- https://github.com/habitat-sh/core-plans/pull/881
+
+A set policy is likely needed because:
+
+- The effort to fix this across all builds is non-trivial
+
+- Some portion of the effort will need to be an acceptance criteria
+  for future core-plan refreshes.
+
+- Eventually `hab-plan-build` integration will be required.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+In the reference-level explanation, a multi-phase implementation
+approach is recommended. This section only provides an explanation for
+the first phase, as the details of the later phases are contingent on
+what we learn from phase 1.
+
+core-plans maintainers and contributors would need to apply hardening
+flags to their plans. This would be done by sourcing a shared script
+in their plan:
+
+    . includes/hardening-defaults.sh
+
+A test will be added to the continuous integration system that checks
```

Comment:
Are we making new packages that are hardened?  In other words, is there is a set that consumers have always used which are non-hardened and then there is a new experimental set that is hardened?

Reply:
My preference would be that we don't split this into two sets.  Rather, we roll these improvements out into the plans that everyone uses.  This is part of the reason I've written up a phased approach and stuck to the set of flags other distributions also widely apply.